### ISSUE TITLE
Openshift Local won't start on Mac

### ISSUE THREAD

**Issue by rkapil** [code-ready/crc#3187]
Hi. I had CRC installed, and then tried to install OpenShift Local on a mac.

1. After installing OpenShift Local (latest for Mac), I did the setup using the GUI. 
2. After setup, I started OSL using the tray app.
3. After a long time, it says it was started.
4. I tried opening the console using the tray comment, but it says peer reset and couldn't connect. oc said the same.
5. I then did the suggested steps on the troubleshooting page (crc delete, cleanup, setup (said everything looks good), then start). 
6. Now it's hung at the log entry attached.

I have 8 core i9 2.3 ghz, 32 GB, running Big Sur. 

Any help is appreciated. Thanks!
[crc.log](https://github.com/code-ready/crc/files/8750335/crc.log)

**Comment by rkapil** [comment]
More info:

CRC version2.3.0+502bf1d9
Red Hat OpenShift Container Platform version4.10.12
Tray version1.2.4

**Comment by praveenkumar** [comment]
@rkapil looking at the logs seems like cluster is started but not become stable, we did get this type of issue recently and our observation is if we give the cluster bit more time it become available. Also please use the issue template which provide more info around the issue.

```
APIServerDeployment_UnavailablePod::CustomRouteController_SyncError::IngressStateEndpoints_NonReadyEndpoints::OAuthServerRouteEndpointAccessibleController_SyncError::OAuthServerServiceEndpointAccessibleController_SyncError::OAuthServerServiceEndpointsEndpointAccessibleController_SyncError"
time="2022-05-22T17:02:02-04:00" level=debug msg="authentication operator not available, Reason: APIServerDeployment_NoPod::APIServices_PreconditionNotReady::OAuthServerRouteEndpointAccessibleController_EndpointUnavailable::OAuthServerServiceEndpointAccessibleController_EndpointUnavailable"
time="2022-05-22T17:02:02-04:00" level=debug msg="console operator is degraded, Reason: CustomRouteSync_FailedDeleteCustomRoutes::RouteHealth_FailedRouteGet::SyncLoopRefresh_FailedGet"
time="2022-05-22T17:02:02-04:00" level=debug msg="console operator not available, Reason: RouteHealth_RouteNotAdmitted"
time="2022-05-22T17:02:02-04:00" level=debug msg="Unexpected operator status for etcd: RecentBackup"
time="2022-05-22T17:02:02-04:00" level=debug msg="image-registry operator not available, Reason: NoReplicasAvailable::NodeCADaemonNoAvailableReplicas"
time="2022-05-22T17:02:02-04:00" level=debug msg="image-registry operator is still progressing, Reason: DeploymentNotCompleted"
time="2022-05-22T17:02:02-04:00" level=debug msg="image-registry operator is degraded, Reason: Unavailable"
time="2022-05-22T17:02:02-04:00" level=debug msg="kube-apiserver operator is degraded, Reason: NodeController_MasterNodesReady"
time="2022-05-22T17:02:02-04:00" level=debug msg="kube-controller-manager operator is degraded, Reason: NodeController_MasterNodesReady"
time="2022-05-22T17:02:02-04:00" level=debug msg="kube-scheduler operator is degraded, Reason: NodeController_MasterNodesReady"
time="2022-05-22T17:02:02-04:00" level=debug msg="openshift-apiserver operator is degraded, Reason: APIServerDeployment_UnavailablePod"
time="2022-05-22T17:02:02-04:00" level=debug msg="openshift-apiserver operator not available, Reason: APIServerDeployment_NoPod::APIServices_PreconditionNotReady"
time="2022-05-22T17:02:02-04:00" level=debug msg="openshift-controller-manager operator is still progressing, Reason: _DesiredStateNotYetAchieved"
time="2022-05-22T17:02:02-04:00" level=debug msg="openshift-controller-manager operator not available, Reason: _NoPodsAvailable"
time="2022-05-22T17:02:02-04:00" level=debug msg="service-ca operator is still progressing, Reason: _ManagedDeploymentsAvailable"
time="2022-05-22T17:02:02-04:00" level=info msg="3 operators are progressing: image-registry, openshift-controller-manager, service-ca"
time="2022-05-22T17:23:06-04:00" level=error msg="Cluster is not ready: cluster operators are still not stable after 21m33.745887955s"
time="2022-05-22T17:23:06-04:00" level=info msg="Adding crc-admin and crc-developer contexts to kubeconfig..."
```

**Comment by gbraad** [comment]
It might help to assign more memory if you have it available...

**Comment by rkapil** [comment]
> @rkapil looking at the logs seems like cluster is started but not become stable, we did get this type of issue recently and our observation is if we give the cluster bit more time it become available. Also please use the issue template which provide more info around the issue.
> 
> ```
> APIServerDeployment_UnavailablePod::CustomRouteController_SyncError::IngressStateEndpoints_NonReadyEndpoints::OAuthServerRouteEndpointAccessibleController_SyncError::OAuthServerServiceEndpointAccessibleController_SyncError::OAuthServerServiceEndpointsEndpointAccessibleController_SyncError"
> time="2022-05-22T17:02:02-04:00" level=debug msg="authentication operator not available, Reason: APIServerDeployment_NoPod::APIServices_PreconditionNotReady::OAuthServerRouteEndpointAccessibleController_EndpointUnavailable::OAuthServerServiceEndpointAccessibleController_EndpointUnavailable"
> time="2022-05-22T17:02:02-04:00" level=debug msg="console operator is degraded, Reason: CustomRouteSync_FailedDeleteCustomRoutes::RouteHealth_FailedRouteGet::SyncLoopRefresh_FailedGet"
> time="2022-05-22T17:02:02-04:00" level=debug msg="console operator not available, Reason: RouteHealth_RouteNotAdmitted"
> time="2022-05-22T17:02:02-04:00" level=debug msg="Unexpected operator status for etcd: RecentBackup"
> time="2022-05-22T17:02:02-04:00" level=debug msg="image-registry operator not available, Reason: NoReplicasAvailable::NodeCADaemonNoAvailableReplicas"
> time="2022-05-22T17:02:02-04:00" level=debug msg="image-registry operator is still progressing, Reason: DeploymentNotCompleted"
> time="2022-05-22T17:02:02-04:00" level=debug msg="image-registry operator is degraded, Reason: Unavailable"
> time="2022-05-22T17:02:02-04:00" level=debug msg="kube-apiserver operator is degraded, Reason: NodeController_MasterNodesReady"
> time="2022-05-22T17:02:02-04:00" level=debug msg="kube-controller-manager operator is degraded, Reason: NodeController_MasterNodesReady"
> time="2022-05-22T17:02:02-04:00" level=debug msg="kube-scheduler operator is degraded, Reason: NodeController_MasterNodesReady"
> time="2022-05-22T17:02:02-04:00" level=debug msg="openshift-apiserver operator is degraded, Reason: APIServerDeployment_UnavailablePod"
> time="2022-05-22T17:02:02-04:00" level=debug msg="openshift-apiserver operator not available, Reason: APIServerDeployment_NoPod::APIServices_PreconditionNotReady"
> time="2022-05-22T17:02:02-04:00" level=debug msg="openshift-controller-manager operator is still progressing, Reason: _DesiredStateNotYetAchieved"
> time="2022-05-22T17:02:02-04:00" level=debug msg="openshift-controller-manager operator not available, Reason: _NoPodsAvailable"
> time="2022-05-22T17:02:02-04:00" level=debug msg="service-ca operator is still progressing, Reason: _ManagedDeploymentsAvailable"
> time="2022-05-22T17:02:02-04:00" level=info msg="3 operators are progressing: image-registry, openshift-controller-manager, service-ca"
> time="2022-05-22T17:23:06-04:00" level=error msg="Cluster is not ready: cluster operators are still not stable after 21m33.745887955s"
> time="2022-05-22T17:23:06-04:00" level=info msg="Adding crc-admin and crc-developer contexts to kubeconfig..."
> ```

Thanks. Can you link the template?

**Comment by rkapil** [comment]
Also, should I be using the tray app or the command line to start/stop the cluster? I just tried starting using the tray app and it does nothing for like 20 min, no log activity so I just stop it. Thanks.

**Comment by praveenkumar** [comment]
> Also, should I be using the tray app or the command line to start/stop the cluster? I just tried starting using the tray app and it does nothing for like 20 min, no log activity so I just stop it. Thanks.

@rkapil you can use tray app or command line. I am suggesting the command line because it is more interactive and provide info to the user as it progress.

>  Can you link the template?

https://github.com/code-ready/crc/issues/new?assignees=&labels=kind%2Fbug%2C+status%2Fneed+triage&template=bug_report.md&title=%5BBUG%5D

**Comment by rkapil** [comment]
### General information

  * OS:macOS 
  * Hypervisor:  vfkit
  * Did you run `crc setup` before starting it? Yes
  * Running CRC on: Laptop 

## CRC version
```bash
# Put `crc version` output here
CRC version: 2.3.0+502bf1d9
OpenShift version: 4.10.12
Podman version: 3.4.4
  
## CRC status
```bash
# Put `crc status --log-level debug` output here
DEBU CRC version: 2.3.0+502bf1d9                  
DEBU OpenShift version: 4.10.12                   
DEBU Podman version: 3.4.4                        
DEBU Running 'crc status'                         
DEBU Checking file: /Users/narayankapil/.crc/machines/crc/.crc-exist 
DEBU Checking file: /Users/narayankapil/.crc/machines/crc/.crc-exist 
DEBU Running SSH command: df -B1 --output=size,used,target /sysroot | tail -1 
DEBU Using ssh private keys: [/Users/narayankapil/.crc/machines/crc/id_ecdsa /Users/narayankapil/.crc/cache/crc_vfkit_4.10.12_amd64/id_ecdsa_crc] 
DEBU SSH command results: err: ssh: handshake failed: read tcp 127.0.0.1:64773->127.0.0.1:2222: read: connection reset by peer, output:  
DEBU Cannot get root partition usage: ssh command error:
command : df -B1 --output=size,used,target /sysroot | tail -1
err     : ssh: handshake failed: read tcp 127.0.0.1:64773->127.0.0.1:2222: read: connection reset by peer 
DEBU cannot get OpenShift status: Get "https://api.crc.testing:6443/apis/config.openshift.io/v1/clusteroperators": read tcp 127.0.0.1:64788->127.0.0.1:6443: read: connection reset by peer 
CRC VM:          Running
OpenShift:       Unreachable (v4.10.12)
Podman:          
Disk Usage:      0B of 0B (Inside the CRC VM)
Cache Usage:     68.48GB
Cache Directory: /Users/narayankapil/.crc/cache

## CRC config
```bash
# Put `crc config view` output here
- consent-telemetry                     : no
- memory                                : 13222

## Host Operating System
```bash
# put the output of `sw_vers` in case of Mac
ProductName:	macOS
ProductVersion:	11.2.1
BuildVersion:	20D75

### Steps to reproduce

  1. Run crc stop, delete, cleanup, setup, start per instructions
  2. Wait for start
  3. Tail crc.log
  4. Wait some more...

### Expected
OpenShift Local to start successfully

### Actual
OpenShift local never starts successfully
Common message is "read tcp 127.0.0.1:64788->127.0.0.1:6443: read: connection reset by peer"


### Logs

Before gather the logs try following if that fix your issue
```bash
$ crc delete -f
$ crc cleanup
$ crc setup
$ crc start --log-level debug
``` DONE

Please consider posting the output of `crc start --log-level debug`  on http://gist.github.com/ and post the link in the issue.
https://gist.github.com/rkapil/3f44d2e5e424bd1c9c2234dd51821c73

**Comment by cfergeau** [comment]
In your initial log, `ssh` connections to the cluster were fine, but the cluster seemed to be taking some time to stabilize.
In your last message, all network connections to the cluster seem to be failing, so I'm not sure we are looking at the same problem.
Can you check if `ssh -i /Users/noname/.crc/machines/crc/id_ecdsa -p 2222 core@127.0.0.1` is working or not? If it fails, can you try to recreate the VM with `crc delete && crc start`?

**Comment by rkapil** [comment]
> ssh -i /Users/noname/.crc/machines/crc/id_ecdsa -p 2222 core@127.0.0.1

It fails with "kex_exchange_identification: read: Connection reset by peer"

Should I run crc cleanup and crc setup or just the two you mentioned?

**Comment by rkapil** [comment]
I installed podman on my mac last week using homebrew, so I'm wondering if that has something to do with this.

**Comment by rkapil** [comment]
After removing podman it's giving the operators messages again but still the connection reset error seems to be the problem. 

ERRO Cannot update kubeconfig: read tcp 127.0.0.1:56240->127.0.0.1:6443: read: connection reset by peer 

Attaching the log. 
[crc.txt](https://github.com/code-ready/crc/files/8757891/crc.txt)

**Comment by rkapil** [comment]
ssh into the cluster worked at this point:  

`time="2022-05-24T06:52:54-04:00" level=info msg="Starting OpenShift cluster... [waiting for the cluster to stabilize]"
.....
mac32:~ noname$ ssh -i /Users/noname/.crc/machines/crc/id_ecdsa -p 2222 core@127.0.0.1
Red Hat Enterprise Linux CoreOS 410.84.202204261500-0
  Part of OpenShift 4.10, RHCOS is a Kubernetes native operating system
  managed by the Machine Config Operator (`clusteroperator/machine-config`).

WARNING: Direct SSH access to machines is not recommended; instead,
make configuration changes via `machineconfig` objects:
  https://docs.openshift.com/container-platform/4.10/architecture/architecture-rhcos.html

---
Last login: Tue May 24 10:51:46 2022 from 192.168.127.1
[core@crc-xhphl-master-0 ~]$ 
`
crc status
`crc status
CRC VM:          Running
OpenShift:       Unreachable (v4.10.12)
Podman:          
Disk Usage:      0B of 0B (Inside the CRC VM)
Cache Usage:     68.48GB
Cache Directory: /Users/noname/.crc/cache`

But then soon afterward the ssh fails when I try to run it again:
`kex_exchange_identification: read: Connection reset by peer`

**Comment by rkapil** [comment]
> @rkapil does any older version of crc worked on this machine or is it first time you are trying CRC?

I had two versions of crc on this machine before trying to install Openshift Local.

**Comment by praveenkumar** [comment]
> > @rkapil does any older version of crc worked on this machine or is it first time you are trying CRC?
> 
> Is there a way I can install an older version? If so, how would I get the pull secret?

@rkapil you can still use older version like 2.2.2 https://github.com/code-ready/crc/releases/tag/v2.2.2 but it will take bit more time to start because the certificate of bundle need to regenerated. I would suggest to try it and see if older version is still working. Also make sure before using older version do cleanup and remove current version.

**Comment by cfergeau** [comment]
Can you add the daemon logs from `~/.crc/crcd.log` after it stops working?

**Comment by cfergeau** [comment]
And can you also check that you have a `vfkit`/`vfkit-amd64` process running when this stops working?

**Comment by rkapil** [comment]
> > > @rkapil does any older version of crc worked on this machine or is it first time you are trying CRC?
> > 
> > 
> > Is there a way I can install an older version? If so, how would I get the pull secret?
> 
> @rkapil you can still use older version like 2.2.2 https://github.com/code-ready/crc/releases/tag/v2.2.2 but it will take bit more time to start because the certificate of bundle need to regenerated. I would suggest to try it and see if older version is still working. Also make sure before using older version do cleanup and remove current version.

When you say remove current version do you mean delete the .crc folder or more than that?

**Comment by jsliacan** [comment]
@rkapil this is a powerful stain removal process:
```
crc cleanup
sudo kill -9 $(pgrep crc-tray | head -1)
sudo rm -rf /Applications/Red\ Hat\ OpenShift\ Local.app/
sudo rm /usr/local/bin/crc
rm -rf ~/.crc/
```
We're using it to have a "clean environment" between tests.

**Comment by rkapil** [comment]
> > > @rkapil does any older version of crc worked on this machine or is it first time you are trying CRC?
> > 
> > 
> > Is there a way I can install an older version? If so, how would I get the pull secret?
> 
> @rkapil you can still use older version like 2.2.2 https://github.com/code-ready/crc/releases/tag/v2.2.2 but it will take bit more time to start because the certificate of bundle need to regenerated. I would suggest to try it and see if older version is still working. Also make sure before using older version do cleanup and remove current version.

The old version started up successfully. Attaching logs.
[crc.txt](https://github.com/code-ready/crc/files/8770431/crc.txt)

**Comment by jsliacan** [comment]
BTW, going from 2.2.2 to 2.3.0 seems to work OK on our machine (macos 12.0.1)

![Screenshot from 2022-05-25 11-38-44](https://user-images.githubusercontent.com/2056138/170257327-cc4e36a9-9867-40b6-948c-aab6bb1d8327.png)

**Comment by shriyo101** [comment]
Seeing the same issue stated above with below version in mac. 

DEBU CRC version: 2.10.1+7e7f6b2d                 
DEBU OpenShift version: 4.11.7                    
DEBU Podman version: 4.2.0  

 ~/.crc $ sw_vers
ProductName:	macOS
ProductVersion:	12.6
BuildVersion:	21G115

So I followed the below steps as suggested in another thread in redhat forum.. can someone tell me why this works?
I would like to understand what went wrong and why giving hosts additional permission & resources works..?

crc config set cpus 8
crc config set memory 16384

sudo chmod 666 /etc/hosts

crc setup
crc start

**Comment by cfergeau** [comment]
@shriyo101 Can you open a new issue and provide details about your setup/what you are trying to do/... ? Thanks!

**Comment by shriyo101** [comment]
Hi Christophe,

My comment was to give an alternative solution that I found useful since
the issue the user had raised was still happening with me.

So I have a Mac m1 and was trying to install openshift crc. Seeing the same
issue stated in Issue #3187 with below version in mac.

DEBU CRC version: 2.10.1+7e7f6b2d
DEBU OpenShift version: 4.11.7
DEBU Podman version: 4.2.0

~/.crc $ sw_vers
ProductName: macOS
ProductVersion: 12.6
BuildVersion: 21G115

So I followed the below steps as suggested in another thread in redhat
forum..

And so I wanted to understand why following the below set of steps works,
because logs dint show much more than what was shared in the thread.
And it didn't make sense when crc is supposed to work with min requirements
but it only worked after giving hosts additional permission & resources.

crc config set cpus 8
crc config set memory 16384

sudo chmod 666 /etc/hosts

crc setup
crc start

I do not face the issue and I no longer have those logs also since I had to
delete and recreate the instance with the new config.

I don't think it needs another thread. But let me know.

Have a good day!

Best Regards,
Yo
